### PR TITLE
Use Node.js CLI to update env vars during start time

### PR DESCRIPTION
**Pull request by paazmaya**
Another option for [#414](https://github.com/Visual-Regression-Tracker/Visual-Regression-Tracker/issues/414)
to answer https://github.com/Visual-Regression-Tracker/frontend/pull/332

### CODE REVIEW

**Comment by pashidlos** [comment]
this should solve security issue https://github.com/Visual-Regression-Tracker/frontend/pull/197/files

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ### STAGE 1: Build ###
-# This image is around 50 megabytes
+# This image is around 53 megabytes
+# https://hub.docker.com/_/node/tags?page=1&name=alpine
```
Comment (pashidlos) [inline Dockerfile:3]: add this part pls 
https://github.com/Visual-Regression-Tracker/frontend/pull/332/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R9-R14

it solves this issue 
https://sonarcloud.io/project/security_hotspots?id=Visual-Regression-Tracker_frontend&hotspots=AYn0nEVqSdwZZQcpatz_

File: Dockerfile [hunk 2]
```diff
@@ -14,19 +15,25 @@ RUN npm ci --verbose
...
-RUN node generate-env-browser.js
-
 ### STAGE 2: Run ###
 # This image is around 5 megabytes
-FROM nginx:1.25-alpine3.17-slim
+# https://hub.docker.com/_/nginx/tags?page=1&name=slim
+FROM nginx:1.25-alpine3.18-slim
+
+# Install Node.js LTS version needed for env variable generation
+RUN apk add --update nodejs
```
Comment (pashidlos) [inline Dockerfile:24]: I guess this part is the reason for image size increase 
https://github.com/Visual-Regression-Tracker/frontend/releases/tag/v5.0.0-rc4
image

Comment (paazmaya) [reply Dockerfile:24]: Yes, it is the smaller possible image that has nodejs

File: Dockerfile [hunk 3]
```diff
@@ -1,32 +1,44 @@
...
 COPY --from=builder /app/build /usr/share/nginx/html
-COPY --from=builder /app/env-config.js /usr/share/nginx/html/
 
-# Nginx server will now start automatically.
+# Environment variables adjust the running environment
+COPY generate-env-browser.js /usr/share/nginx/html/generate-env-browser.mjs
+COPY .env .
+
+# Start Nginx server with override ENV variables script
+CMD ["/bin/sh", "-c", "node /usr/share/nginx/html/generate-env-browser.mjs && nginx -g \"daemon off;\""]
```
Comment (pashidlos) [inline Dockerfile:44]: could you try with docker-compose if it works?
I get error using those versions
VRT_UI_VERSION=5.0.0-rc4
VRT_API_VERSION=5.0.0
VRT_MIGRATION_VERSION=5.0.0

image

**Pull request closed without merging**